src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -273,6 +272,27 @@ void RegisterContext::DispatchSuccess()
     callback(context, typeWithoutSubTypes.c_str(), CHIP_NO_ERROR);
 }
 
+RegisterRecordContext::RegisterRecordContext(DnssdPublishCallback cb, void * cbContext)
+{
+    type     = ContextType::RegisterRecord;
+    context  = cbContext;
+    callback = cb;
+}
+
+void RegisterRecordContext::DispatchFailure(DNSServiceErrorType err)
+{
+    ChipLogError(Discovery, "Mdns: Register Record Failure (%s)", Error::ToString(err));
+    // Return CHIP_ERROR_MDNS_COLLISION error so that the caller can handle a name conflict
+    CHIP_ERROR error = (err == kDNSServiceErr_NameConflict) ? CHIP_ERROR_MDNS_COLLISION : CHIP_ERROR_INTERNAL;
+    callback(context, nullptr, error);
```

Comment:
What about adding some code to `MdnsError.cpp` to start mapping dns error to chip error better ?

```
CHIP_ERROR ToChipError(DNSServiceErrorType errorCode)
{
   switch (errorCode)
   {
     case kDNSServiceErr_NoError:
        return CHIP_NO_ERROR;
     case kDNSServiceErr_NameConflict:
        return CHIP_ERROR_MDNS_COLLISION;
      default:
        return CHIP_ERROR_INTERNAL;
   }
}
```

This way those lines can be rewritten as `callback(context, nullptr, Error::ToChipError(err));` and if needed it can be used in other places too to have a better understanding of the error at the end of the day.

Reply:
done